docs/src/apps/jsonrpc-api.md

```diff
@@ -395,12 +396,34 @@ The signatures will be ordered based on the Slot in which they were confirmed in
 
 ```bash
 // Request
-curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc": "2.0","id":1,"method":"getConfirmedSignaturesForAddress","params":["6H94zdiaYfRfPfKjYLjyr2VFBg6JHXygy84r3qhc3NsC", 0, 100]}' localhost:8899
+curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc": "2.0","id":1,"method":"deprecatedGetConfirmedSignaturesForAddress","params":["6H94zdiaYfRfPfKjYLjyr2VFBg6JHXygy84r3qhc3NsC", 0, 100]}' localhost:8899
 
 // Result
 {"jsonrpc":"2.0","result":{["35YGay1Lwjwgxe9zaH6APSHbt9gYQUCtBWTNL3aVwVGn9xTFw2fgds7qK5AL29mP63A9j3rh8KpN1TgSR62XCaby","4bJdGN8Tt2kLWZ3Fa1dpwPSEkXWWTSszPSf1rRVsCwNjxbbUdwTeiWtmi8soA26YmwnKD4aAxNp8ci1Gjpdv4gsr","4LQ14a7BYY27578Uj8LPCaVhSdJGLn9DJqnUJHpy95FMqdKf9acAhUhecPQNjNUy6VoNFUbvwYkPociFSf87cWbG"]},"id":1}
 ```
 
+### getConfirmedSignaturesForAddress
+
+Returns confirmed signatures for transactions involving an
+address backwards in time from the provided slot
+
+#### Parameters:
+* `<string>` - account address as base-58 encoded string
+* `<object>` - (optional) Configuration object containing the following field:
+  * `startAfter: <string>` - start searching backwards from this transaction signature,
+                             which must be a confirmed signature for the account
+                             address.  If not provided the search starts from
+                             the highest max confirmed block.
+  * `limit: <number>` - maximum transaction signatures to return (between 1 and 1,000, default: 1,000).
+
+#### Results:
+The result field will be an array of:
+* `<string>` - transaction signature as base-58 encoded string, ordered from newest to oldest.
```

Comment:
While we're at it, it would be nice to include the status and slot number for each signature.

Reply:
I agree it would be nice for some users.   Looking up the status and slot number requires significantly more work for the RPC server and I'd like to not do that work by default.   

Some options:
1. Do nothing.  Make the client shuttle the signatures back to the RPC node via `getSignatureStatus` (which hopefully will be a smaller set of signatures than what's returned here)
2. Add an optional input configuration option.  If we do this then I'd want to reduce the number of signatures returned here from 1,000 to 256 to match the `getSignatureStatus`  bounds

Reply:
Doing nothing sounds fine to me, didn't realize it would be significantly more work to pull that info.

Reply:
It might still be possible actually, with a but of restructuring of the storage backend.   `Option<TransactionError>` and a`Slot` isn't that big, relative to the signature itself.   What do you think the user would do with that info though?  If they'd just turn around and call `getSignatureStatus` anyway then there's probably no benefit.

Reply:
> If they'd just turn around and call getSignatureStatus anyway then there's probably no benefit

The reason I brought it up was to elide the call to `getSignatureStatus` altogether. Beyond status and slot, there's just confirmations which isn't relevant for confirmed signatures.

> What do you think the user would do with that info though?

I'm not sure what all the use cases are but for the explorer I felt that signatures by themselves were pretty meaningless. Seeing status and slot adds a lot of context (status for checking why a tx failed and slot for how recently active an account was)

![Screen Shot 2020-07-08 at 11 07 43 PM](https://user-images.githubusercontent.com/1076145/86935820-d9555400-c16f-11ea-8f55-93997f704b30.png)

Reply:
I think we can do this.  I'm re-structuring the S3 data (need to update #10900 still) to hold this additional information.

Reply:
I updated the RPC response to include the slot/status